src/schemas/clinical-mutations/clearRegistration.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { operations } from '../../submission/submission-service';
+
+const clearClinicalRegistrationMutation = {
+  clearClinicalRegistration: async (
+    obj: unknown,
+    args: {
+      shortName: string;
+      registrationId: string;
+    },
+  ) => {
+    const { shortName, registrationId } = args;
+    const response = await operations.deleteRegistration(registrationId, shortName);
```

Comment:
It would be beneficial to handle errors instead of always returning `true`
service layer can throw an error and repo layer can throw an error

Reply:
I've added resolve/reject handling here with error handling, let me know how it looks to you

Reply:
Should probably throw instead of just logging?

Reply:
for my own learning , is there something specific about GQL that the spec asks you to return a boolean? Or is it something with our system?

Reply:
It's how our mutation is defined:
https://github.com/icgc-argo/argo-clinical/blob/86a60d12dfdf4dde3853c68c61cdfea2a1a5878b/src/schemas/gqlTypeDefs.ts#L44

So any incoming requests are expecting to receive a boolean back; changing this necessitates changing the requests